observer/circularQueueNodesProvider.go

```diff
@@ -35,11 +35,11 @@ func NewCircularQueueNodesProvider(observers []*data.NodeData, configurationFile
 }
 
 // GetNodesByShardId will return a slice of observers for the given shard
-func (cqnp *circularQueueNodesProvider) GetNodesByShardId(shardId uint32) ([]*data.NodeData, error) {
+func (cqnp *circularQueueNodesProvider) GetNodesByShardId(shardId uint32, dataAvailability data.ObserverDataAvailabilityType) ([]*data.NodeData, error) {
 	cqnp.mutNodes.Lock()
```

Comment:
RLock ?

Reply:
since I also perform write operations, I would leave it as it is